File: 9.palindrome-number/solution.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0)
        	return false;
        String s = String.valueOf(x);
        int len = s.length();
        int half = len / 2;
        for (int i = 0; i < half; i++)
        {
        	if (s.charAt(i) != s.charAt(len - 1 - i))
        		return false;
        }
        return true;
    }
}

File: 9.palindrome-number/solution.go
func isPalindrome(x int) bool {
	if x <= 0 {
		return x == 0
	}
	xx := int64(x)
	var a []int64
	for xx > 0 {
		a = append(a, xx%10)
		xx /= 10
	}
	y := int64(0)
	for _, v := range a {
		y = y*10 + v
	}
	return y == int64(x)
}
